libs/solvers/unittests/test_adapt.cpp

```diff
@@ -12,37 +12,179 @@
 #include "cudaq.h"
 #include "nvqpp/test_kernels.h"
 #include "cudaq/solvers/adapt.h"
+#include "cudaq/solvers/operators.h"
 
-std::vector<double> h2_data{
-    3, 1, 1, 3, 0.0454063,  0,  2, 0, 0, 0, 0.17028,    0,
-    0, 0, 2, 0, -0.220041,  -0, 1, 3, 3, 1, 0.0454063,  0,
-    0, 0, 0, 0, -0.106477,  0,  0, 2, 0, 0, 0.17028,    0,
-    0, 0, 0, 2, -0.220041,  -0, 3, 3, 1, 1, -0.0454063, -0,
-    2, 2, 0, 0, 0.168336,   0,  2, 0, 2, 0, 0.1202,     0,
-    0, 2, 0, 2, 0.1202,     0,  2, 0, 0, 2, 0.165607,   0,
-    0, 2, 2, 0, 0.165607,   0,  0, 0, 2, 2, 0.174073,   0,
-    1, 1, 3, 3, -0.0454063, -0, 15};
-
-TEST(SolversTester, checkSimpleAdapt) {
-  cudaq::spin_op h(h2_data, 4);
+bool check_gpu_available() {
+  static int gpu_available = -1; // -1 means unknown; 0 = no; 1 = yes
+  if (gpu_available < 0) {
+    if (std::system("nvidia-smi > /dev/null 2>&1") != 0)
+      gpu_available = 0;
+    else
+      gpu_available = 1;
+  }
+  return static_cast<bool>(gpu_available);
+}
+
+class SolversTester : public ::testing::Test {
+protected:
+  cudaq::spin_op h;
+  cudaq::spin_op hamli;
+  cudaq::spin_op hamhh;
+  void SetUp() override {
```

Comment:
As I understand it, this method is called at the beginning of each test, but it looks like it only needs to be run once as far as I can tell. Perhaps it could be sped up with this change.
```suggestion
  void SetUp() override {
    if (!h.is_identity())
      return; // early return if we're already setup
```

Reply:
Unfortunately the approach was not working. I used the proper of doing it in gtest :)